### ISSUE TITLE
Document inheritance and mixins

### ISSUE THREAD

**Issue by Sinjhin** [canjs/canjs#3008]
justinmeyer
so the only thing you might not see is can.Construct
is that what you mean?
b/c can.Component, can.Map, etc inherit from that?
otherwise, everything comes directly from the module

john
Yeah, but also there are other links. Like how can-define gets all of can-event’s methods.
I guess being able to visualize that would be helpful.

justinmeyer
yeah, the mixins
I think that could be helpful
we probably should list that can-define has `addEventListener` and such mixed in ...

**Comment by justinbmeyer** [comment]
@chasenlehara, @Sinjhin shared: https://developer.android.com/reference/android/app/Application.html

Notice how it shows the inheritance chain and interfaces it implements:

<img width="528" alt="cursor_and_application___android_developers" src="https://cloud.githubusercontent.com/assets/78602/22812228/295a68a6-ef09-11e6-81f3-d935c7524078.png">


We should have something similar.   We probably need `mixins` as well as `interfaces` a module might implement.

**Comment by chasenlehara** [comment]
Yeah, good idea. If we can define mixins well, we can do something similar to how Apple handles protocols:
https://developer.apple.com/reference/uikit/uicollectionviewtransitionlayout#relationships
<img width="556" alt="screen shot 2017-02-10 at 3 11 42 pm" src="https://cloud.githubusercontent.com/assets/10070176/22847681/6547044c-efa3-11e6-813e-dc0255ef56ce.png">

**Comment by Sinjhin** [comment]
@justinbmeyer Another example that came up for me today and seems to be totally undocumented.

The events prop of a component apparently is a `can-control` and has an `off()` that is no-where in the docs?

**Comment by justinbmeyer** [comment]
@Sinjhin It does say events is used as the prototype of a `can-control`: http://canjs.com/doc/can-component.prototype.events.html 


Not every method needs to be documented.  `off()` is more of an internal method, not expected for other people to call.
